### PR TITLE
Fix/refresh with scope

### LINKED ISSUE #373: Cannot set scope on refresn token
this:
```
await _oidcClient.RefreshTokenAsync(entity.RefreshToken, new Parameters(new Dictionary<string, string>
        {
            ["scope"] = scope
        })
```

throw "InvalidOperationException : Duplicate parameter: scope" and we can set a scope when refreshing a token

### PR DESCRIPTION
**Pull request by aguacongas**
fix #373

### CODE REVIEW

File: src/OidcClient/OidcClient.cs [hunk 1]
```diff
@@ -313,13 +313,15 @@ public virtual async Task<UserInfoResult> GetUserInfoAsync(string accessToken, C
...
+        /// <param name="scope">Space separated list of the requested scopes.</param>
         /// <returns>
         /// A token response.
         /// </returns>
         public virtual async Task<RefreshTokenResult> RefreshTokenAsync(
             string refreshToken, 
             Parameters backChannelParameters = null, 
-            CancellationToken cancellationToken = default)
+            CancellationToken cancellationToken = default,
+            string scope = null)
```
Comment (leastprivilege) [inline src/OidcClient/OidcClient.cs]: Could you make scope the 3rd parameter?

**Review by leastprivilege — CHANGES_REQUESTED** [review]

File: src/OidcClient/OidcClient.cs (hunk 1 above)
Comment (aguacongas) [reply src/OidcClient/OidcClient.cs]: It introduces a breaking change then?

Comment (leastprivilege) [reply src/OidcClient/OidcClient.cs]: It does already...

**Review by leastprivilege — APPROVED** [review]

**Comment by leastprivilege** [comment]
thanks!

File: src/OidcClient/OidcClient.cs (hunk 1 above)
Comment (ashuvviet) [reply src/OidcClient/OidcClient.cs]: @leastprivilege  this change breaking the functionally from interceptor i.e. RefreshTokenDelegatingHandler when token is refreshed but scope is default is null and pass from options. due to that refresh token always fails.

Comment (ashuvviet) [reply src/OidcClient/OidcClient.cs]: fix could be :

var response = await _oidcClient.RefreshTokenAsync(_refreshToken, null, _oidcClient.Options.Scope, cancellationToken: cancellationToken)

**Pull request merged**